File: FrontEnd/src/model/tutor.ts
import { Account } from "./auth";


export interface Tutor extends Account {
  id?: number;
  name?: string;
  town?: Town;
  university?: University;
  speciality?: Speciality;
  mail?: string;
  subjects?: Subject[];
  hourlyPrice?: number;
  averageRate?: number;
  rateCount?: number;
  photoUrlId?: number;
}

export enum Town {
  Tartu = "Tartu",
  Tallinn = "Tallinn",
  Narva = "Narva",
  KohtlaJarve = "Kohtla-Järve",
}

export enum University {
  UniversityOfTartu = "University of Tartu",
  TallinnTechnicalUniversity = "Tallinn Technical University",
  TallinnUniversity = "Tallinn University",
  TallinnCollegeOfHealth = "Tallinn College Of Health",
  TartuHigherArtSchool = "Tartu Higher Art School",
}

export enum Speciality {
  BusinessIt = "Business It",
  Economics = "Economics",
  Informatics = "Informatics",
  Science = "Science",
  CyberSecurity = "Cyber Security",
  Psychologist = "Psychologist",
}

export enum Subject {
  Economics = "Economics",
  Maths = "Maths",
  Programming = "Programming",
  Startup = "Startup",
  PE = "PE",
  DiscMaths = "DiscMaths",
}

export interface State {
  Tutors: Tutor[];
}

File: BackEnd/Model/Tutor.cs
using System.Text.Json.Serialization;
using backend.Model;
using Microsoft.EntityFrameworkCore;

namespace tuutoriplatvorm.Model
{
    public class Tutor
    {
        public int? Id { get; set; }
        [JsonIgnore]
        public string? Username { get; set; }
        public required string Name { get; init; }
        public Town? Town { get; init; }
        public University? University { get; init; }
        public Speciality? Speciality { get; init; }
        public string? Mail { get; init; }
        public required List<Subject> Subjects { get; set; }
        public double? HourlyPrice { get; init; }
        public decimal? AverageRate { get; set; }
        public int? RateCount { get; set; }
        public string? PhotoUrlId { get; set; }

        public  int? ScheduleId { get; set; } 

        [JsonIgnore]
        public ICollection<Schedule> Schedules { get; } = new List<Schedule>();
        [JsonIgnore]
        public virtual List<StudentRateTutor> StudentRateTutors { get; set; } = new();

    }


    [JsonConverter(typeof(JsonStringEnumConverter))]
    public enum Speciality
    {
        BusinessIt,
        Economics,
        Informatics,
        Science,
        CyberSecurity,
        Psychologist,

    }
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public enum University
    {
        UniversityOfTartu = 1,
        TallinnTechnicalUniversity = 2,
        TallinnUniversity = 3,
        TallinnCollegeOfHealth = 4,
        TartuHigherArtSchool = 5,
    }
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public enum Town
    {
        Tartu = 1,
        Tallinn = 2,
        Narva = 3,
        KohtlaJarve = 4,

    }
}